app/scripts/directives/overview/mobileClientRow.js

```diff
@@ -44,26 +92,65 @@
             break;
         }
       }
-    };
 
+      if (apiObjectChanges && !row.context) {
+        row.context = {namespace: _.get(row, 'apiObject.metadata.namespace')};
+      }
+
+      if (apiObjectChanges && !row.serviceInstancesWatched) {
+        row.serviceInstancesWatched = true;
+        DataService.list(serviceClassesVersion, row.context, function(serviceClasses) {
+          serviceClasses = serviceClasses.by('metadata.name');
+          watches.push(DataService.watch(serviceInstancesVersion, row.context, function(serviceinstances) {
+            row.services = _.filter(serviceinstances.by('metadata.name'), function(serviceInstance){
+              var serviceClass = _.get(serviceClasses, ServiceInstancesService.getServiceClassNameForInstance(serviceInstance));
+              return isMobileService(serviceClass) && isServiceInstanceReady(serviceInstance);
+            });
+            row.mobileClientEnabledServices = _.filter(row.services, function(service){
+              var serviceClass = _.get(serviceClasses, ServiceInstancesService.getServiceClassNameForInstance(service));
+              return isMobileClientEnabled(serviceClass);
+            });
+            row.updateServicesInfo();
+          }, { errorNotification: false }));
+        });
+      }
+
+      if (apiObjectChanges && !row.bindingsWatched) {
+        row.bindingsWatched = true;
+        watches.push(DataService.watch(serviceBindingsVersion, row.context, function(bindingsData) {
+          row.bindings = _.filter(bindingsData.by('metadata.name'), function(binding) {
+            return _.get(binding.metadata.annotations, 'binding.aerogear.org/consumer') === _.get(row, 'apiObject.metadata.name') && isBindingReady(binding);
+          });
+          row.updateServicesInfo();
+        }));
+      }
 
-    row.mobileclientVersion = {
-      group: "mobile.k8s.io",
-      version: "v1alpha1",
-      resource: "mobileclients"
+      if (apiObjectChanges && !row.buildsWatched) {
+        row.buildsWatched = true;
+        watches.push(DataService.watch(buildsVersion, row.context, function(buildsData) {
+          // Filter builds by mobile client id
+          var builds = _.filter(buildsData.by('metadata.name'), function(build) {
+            return _.get(build, 'metadata.labels.mobile-client-id') === _.get(row, 'apiObject.metadata.name');
+          });
+          row.builds = BuildsService.sortBuilds(builds, true);
+        }));
+      }
     };
+
     row.actionsDropdownVisible = function () {
-      // no actions on those marked for deletion
       if (_.get(row.apiObject, 'metadata.deletionTimestamp')) {
```

Comment:
Rather than a `return false`, I'd prob just do:

```JavaScript
return !!_.get(row.apiObject, 'metadata.deletionTimestamp') ||
  AuthorizationService.canI(row.mobileClientsVersion, 'delete');
```
Its a little more idiomatic.